CRM/Member/Form/MembershipRenewal.php

```diff
@@ -339,6 +339,12 @@ public function buildQuickForm() {
 
     $this->applyFilter('__ALL__', 'trim');
 
+    // CRM-20571
+    // Get the Join Date from Membership info as it is not available in the Renew form
+    $joinDate = CRM_Core_DAO::getFieldValue('CRM_Member_DAO_Membership', $this->_id, 'join_date');
+    // Save its value in a hidden field
+    $this->addElement('hidden', 'join_date', $joinDate);
```

Comment:
Can we do this validation using `addRule()` in `buildQuickForm()` itself. I didn't find any documentation on what kind of complex parameters can `addRule()` take. Is there a way to compare two dates in `addRule()`. The best complexity I saw is this:
```
 $this->addRule('trxn_id', ts('Transaction ID already exists in Database.'),
        'objectExists', array('CRM_Contribute_DAO_Contribution', $this->_id, 'trxn_id')
      );
```